Samples/OpenQasm/OpenQasmDriver.cs

```diff
@@ -0,0 +1,321 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Quantum.Primitive;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Microsoft.Quantum.Samples.OpenQasm
+{
+    /// <summary>
+    /// Quick and dirty Simulatorbase to write OpenQASM 2.0
+    /// Just enough to show that it would work
+    /// Please don't put this in production until its fully engineerd.
+    /// This code could eat your cat. So imagine what schrodinger has to say about that one.
+    /// </summary>
+    public abstract class OpenQasmDriver : SimulatorBase
+    {
+        private int operationDepth;
+        public override void StartOperation(string operationName, OperationFunctor functor, object inputValue)
```

Comment:
The StartOperation and EndOperation are the only things I have a bit of concerned. On the upcoming version we will change the interface of these methods to receive different parameters, and the behavior will also change as today Primitive operations (like X or H) are not triggering these events, but now they will, and it feels to me this will break this code.
Could you instead use the constructor or the `RunOpenQasm` method instead?
If primitive operations triggering these methods is not a problem, we could leave it like this and I will just take care of updating the signature with the new version, or even better, we could just wait for the new version which should not take long.

Reply:
Removed the dependency on StartOperation and EndOperation and put it in the constructor. I put it in the RunOpenQasm, I will need to either wrap it all the time when there are multiple measurements in there, or check if it already has been added. Next version needs to measure more qubits, so a single action can be re-used.